modules/mono/editor/GodotTools/GodotTools.IdeMessaging/Client.cs

```diff
@@ -121,6 +121,7 @@ public Client(string identity, string godotProjectDir, IMessageHandler messageHa
             this.messageHandler = messageHandler;
             this.logger = logger;
 
+            // TODO: Need to fetch the project data dir name from ProjectSettings instead of defaulting to ".godot"
```

Comment:
I think that'd be something like:
```csharp
string godotProjectDataDir = ProjectSettings.GetProjectDataPath();
string projectMetadataDir = Path.Combine(godotProjectDataDir, "mono", "metadata");
```

Didn't test though.

Reply:
That won't work. The `GodotTools.IdeMessaging` project does not reference the `GodotSharp.dll` which contains the `Godot.ProjectSettings` class. Also, the `GetProjectDataPath` returns a path that starts with `res://` but `projectMetadataDir` needs to be a proper absolute path.

I think this library is supposed to be usable without Godot (it's used by the VSCode extension) so adding a reference to `GodotSharp.dll` is not desired, but I could be wrong.

Reply:
Any idea for a potential solution @neikeq?

Otherwise this is one example of why making the project data path configurable can be tricky, as it prevents external tooling from knowing what the path is unless they parse `project.godot` and reimplement the logic themselves (or they need to depend on Godot itself).

Reply:
Let's keep it hardcoded for now, there's a TODO so it can be resolved later. But if #52548 gets actually implemented and used, this will become a bug (so we should probably file an issue for this TODO).

Reply:
Sorry, I missed this. Yes, this is an example of why I think making the path configurable would be a bad idea. The only option for this library would be to either hard-code the path or parse the `project.godot` file which is not easy as it's not a standard file format.